### PR TITLE
Comment on generators that won't be following new_opmath

### PR DESCRIPTION
**Pull request by lillian542**
After looking at the tests for a bit and realizing what was going on, I would recommend these generators return new_opmath operators regardless of whether new_opmath is installed, because the new_opmath was explicitly used to add functionality that isn't available with `qml.Hamiltonian`:

- the "Identity with no wires" functionality only works with new opmath, and is needed for `ControlledGlobalPhase`. Changing `Identity` and `Controlled` is just undoing #5194. 
- a Hamiltonian is not suitable for the qutrit operators, because it can't return a matrix for a qutrit system, breaking parameter frequencies

I've added a comment on each of the relevant generator functions, since this will differ from the standard format, and that's all.

### CODE REVIEW

File: pennylane/ops/identity.py [hunk 1]
```diff
@@ -412,4 +412,4 @@ def pow(self, z):
         return [GlobalPhase(z * self.data[0], self.wires)]
 
     def generator(self):
-        return qml.s_prod(-1, qml.I(self.wires))
+        return qml.Hamiltonian([-1], [qml.I(self.wires)])
```
Comment (lillian542) [inline pennylane/ops/identity.py:415]:
```suggestion
        return -1 * [qml.I(self.wires)]
```

File: pennylane/ops/identity.py [hunk 2]
```diff
@@ -412,4 +412,6 @@ def pow(self, z):
         return [GlobalPhase(z * self.data[0], self.wires)]
 
     def generator(self):
-        return qml.s_prod(-1, qml.I(self.wires))
+        # needs to return a new_opmath instance regardless of whether new_opmath is enabled, because
+        # it otherwise can't handle Identity with no wires, see PR #5194
+        return qml.s_prod(-1, qml.Identity(self.wires))
```
Comment (lillian542) [inline pennylane/ops/identity.py:417]:
```suggestion
        return qml.s_prod(-1, qml.I(self.wires))
```

**Comment by lillian542** [comment]
[sc-57982]

**Review by PietropaoloFrisoni — APPROVED** [review]
Yep, I remember having the same issue with these a few weeks ago. I am fine with them the way they are (maybe we should create a story for the future).

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -139,9 +139,10 @@
 * The `molecular_hamiltonian` function calls `PySCF` directly when `method='pyscf'` is selected.
   [(#5118)](https://github.com/PennyLaneAI/pennylane/pull/5118)
 
-* All generators in the source code (except those in the `qchem` module) no longer return
-  `Hamiltonian` or `Tensor` instances. Wherever possible, these return `Sum`, `SProd`, and `Prod` instances.
+* The generators in the source code return operators consistent with the global setting for 
+  `qml.operator.active_new_opmath()` whereever possible. `Sum`, `SProd` and `Prod` instances will be returned even after disabling the new operator arithmetic in cases where they offer additional functionality not available using legacy operators.
   [(#5253)](https://github.com/PennyLaneAI/pennylane/pull/5253)
+  [(#5412)](https://github.com/PennyLaneAI/pennylane/pull/5412)
```
Comment (lillian542) [inline doc/releases/changelog-dev.md:145]:
```suggestion
  [(#5411)](https://github.com/PennyLaneAI/pennylane/pull/5411)  
```

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**